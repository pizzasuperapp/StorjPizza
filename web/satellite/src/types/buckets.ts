// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all bucket-related functionality.
 */
export interface BucketsApi {
    /**
     * Fetch buckets
     *
     * @returns BucketPage
     * @throws Error
     */
    get(projectId: string, before: Date, cursor: BucketCursor): Promise<BucketPage>;

    /**
     * Fetch all bucket names
     *
     * @returns string[]
     * @throws Error
     */
    getAllBucketNames(projectId: string): Promise<string[]>;
}

/**
 * Bucket class holds info for Bucket entity.
 */
export class Bucket {
    public constructor(
        public name: string = '',
        public storage: number = 0,
        public egress: number = 0,
        public objectCount: number = 0,
        public segmentCount: number = 0,
        public since: Date = new Date(),
        public before: Date = new Date(),
    ) { }
}

/**
 * BucketPage class holds bucket total usages and flag whether more usages available.
 */
export class BucketPage {
    public constructor(
        public buckets: Bucket[] = [],
        public search: string = '',
        public limit: number = 0,
        public offset: number = 0,
        public pageCount: number = 0,
        public currentPage: number = 0,
        public totalCount: number = 0,
    ) { }
}

/**
 * BucketCursor class holds cursor for bucket name and limit.
 */
export class BucketCursor {
    public constructor(
        public search: string = '',
        public limit: number = 0,
        public page: number = 0,
    ) { }
}
