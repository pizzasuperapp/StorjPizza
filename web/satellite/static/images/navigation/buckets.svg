<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M16.3549 7.54666C17.762 8.95376 17.7748 11.227 16.3778 12.624C15.8117 13.1901 15.1018 13.5247 14.3639 13.6289L14.1106 15.0638C14.092 16.1897 11.7878 17.1 8.94808 17.1C6.12541 17.1 3.83185 16.2006 3.78626 15.0841L3.78583 15.0638L1.84465 4.06597C1.8228 3.98819 1.80882 3.90946 1.80303 3.82992L1.79999 3.81221L1.80189 3.81231C1.80062 3.79028 1.79999 3.76818 1.79999 3.74602C1.79999 2.17422 5.0003 0.900024 8.94808 0.900024C12.8959 0.900024 16.0962 2.17422 16.0962 3.74602C16.0962 3.76818 16.0955 3.79028 16.0943 3.81231L16.0962 3.81221L16.093 3.83111C16.0872 3.90975 16.0734 3.98759 16.0519 4.06451L15.5749 6.76662L16.3549 7.54666ZM14.2961 5.63437C12.9867 6.22183 11.076 6.59202 8.94808 6.59202C6.82026 6.59202 4.90959 6.22185 3.60014 5.63443L5.00723 13.6077L5.23729 14.8286L5.25861 14.8452C5.37892 14.9354 5.56514 15.0371 5.80696 15.1351L5.85606 15.1546C6.63551 15.4594 7.74619 15.6439 8.94808 15.6439C10.1569 15.6439 11.2733 15.4573 12.0527 15.1497C12.3337 15.0388 12.5431 14.9223 12.6661 14.8231L12.6761 14.8148L12.9019 13.5348C12.3338 13.3787 11.7955 13.0812 11.3429 12.6429L11.3004 12.6011L8.37488 9.67559C8.09056 9.39127 8.09056 8.93029 8.37488 8.64597C8.65226 8.36859 9.09779 8.36182 9.38338 8.62568L9.40449 8.64597L12.33 11.5715C12.5717 11.8132 12.8555 11.9861 13.1569 12.0901L14.2961 5.63437ZM15.2661 8.51698L14.6408 12.0597C14.8989 11.9575 15.1402 11.8024 15.3482 11.5944C16.1641 10.7784 16.1664 9.44942 15.3549 8.60652L15.3253 8.57627L15.2661 8.51698ZM8.94808 2.35612C7.20125 2.35612 5.58125 2.62893 4.43223 3.08641C3.93851 3.28298 3.57117 3.49947 3.35976 3.69848C3.34629 3.71116 3.33399 3.72325 3.32283 3.73469L3.31221 3.74589L3.33142 3.76606L3.35976 3.79357C3.57117 3.99258 3.93851 4.20906 4.43223 4.40564C5.58125 4.86312 7.20125 5.13593 8.94808 5.13593C10.6949 5.13593 12.3149 4.86312 13.4639 4.40564C13.9576 4.20906 14.325 3.99258 14.5364 3.79357C14.5499 3.78089 14.5622 3.7688 14.5733 3.75735L14.5841 3.74589L14.5647 3.72599L14.5364 3.69848C14.325 3.49947 13.9576 3.28298 13.4639 3.08641C12.3149 2.62893 10.6949 2.35612 8.94808 2.35612Z" fill="#56606D"/>
</svg>
