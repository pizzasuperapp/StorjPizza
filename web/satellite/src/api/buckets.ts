// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { BaseGql } from '@/api/baseGql';
import { ErrorUnauthorized } from '@/api/errors/ErrorUnauthorized';
import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import { HttpClient } from '@/utils/httpClient';

/**
 * BucketsApiGql is a graphql implementation of Buckets API.
 * Exposes all bucket-related functionality.
 */
export class BucketsApiGql extends BaseGql implements BucketsApi {
    private readonly client: HttpClient = new HttpClient();
    private readonly ROOT_PATH: string = '/api/v0/buckets';

    /**
     * Fetch buckets.
     *
     * @returns BucketPage
     * @throws Error
     */
    public async get(projectId: string, before: Date, cursor: BucketCursor): Promise<BucketPage> {
        const query =
            `query($projectId: String!, $before: DateTime!, $limit: Int!, $search: String!, $page: Int!) {
                project(id: $projectId) {
                    bucketUsages(before: $before, cursor: {
                        limit: $limit, search: $search, page: $page
                    }) {
                        bucketUsages {
                            bucketName,
                            storage,
                            egress,
                            objectCount,
                            segmentCount,
                            since,
                            before
                        },
                        search,
                        limit,
                        offset,
                        pageCount,
                        currentPage,
                        totalCount
                    }
                }
            }`;

        const variables = {
            projectId,
            before: before.toISOString(),
            limit: cursor.limit,
            search: cursor.search,
            page: cursor.page,
        };

        const response = await this.query(query, variables);

        return this.getBucketPage(response.data.project.bucketUsages);
    }

    /**
     * Fetch all bucket names.
     *
     * @returns string[]
     * @throws Error
     */
    public async getAllBucketNames(projectId: string): Promise<string[]> {
        const path = `${this.ROOT_PATH}/bucket-names?projectID=${projectId}`;
        const response = await this.client.get(path);

        if (!response.ok) {
            if (response.status === 401) {
                throw new ErrorUnauthorized();
            }

            throw new Error('Can not get bucket names');
        }

        const result = await response.json();

        return result ? result : [];
    }

    /**
     * Method for mapping buckets page from json to BucketPage type.
     *
     * @param page anonymous object from json
     */
    private getBucketPage(page: any): BucketPage { // eslint-disable-line @typescript-eslint/no-explicit-any
        if (!page) {
            return new BucketPage();
        }

        const buckets: Bucket[] = page.bucketUsages.map(key =>
            new Bucket(
                key.bucketName,
                key.storage,
                key.egress,
                key.objectCount,
                key.segmentCount,
                new Date(key.since),
                new Date(key.before)));

        return new BucketPage(buckets, page.search, page.limit, page.offset, page.pageCount, page.currentPage, page.totalCount);
    }
}
