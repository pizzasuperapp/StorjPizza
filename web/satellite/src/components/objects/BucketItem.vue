// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="bucket-item">
        <div class="bucket-item__name">
            <bucket-icon />
            <p class="bucket-item__name__value">{{ itemData.name }}</p>
        </div>
        <p class="bucket-item__date">{{ formattedDate }}</p>
        <div v-click-outside="closeDropdown" class="bucket-item__functional" @click.stop="openDropdown(dropdownKey)">
            <dots-icon />
            <div v-if="isDropdownOpen" class="bucket-item__functional__dropdown">
                <div class="bucket-item__functional__dropdown__item" @click.stop="onDetailsClick">
                    <details-icon />
                    <p class="bucket-item__functional__dropdown__item__label">View Bucket Details</p>
                </div>
                <div class="bucket-item__functional__dropdown__item" @click.stop="onDeleteClick">
                    <delete-icon />
                    <p class="bucket-item__functional__dropdown__item__label">Delete Bucket</p>
                </div>
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import BucketIcon from '@/../static/images/objects/bucketItem.svg';
import DeleteIcon from '@/../static/images/objects/delete.svg';
import DetailsIcon from '@/../static/images/objects/details.svg';
import DotsIcon from '@/../static/images/objects/dots.svg';
import {RouteConfig} from "@/router";
import {Bucket} from "@/types/buckets";

// @vue/component
@Component({
    components: {
        BucketIcon,
        DotsIcon,
        DeleteIcon,
        DetailsIcon,
    },
})
export default class BucketItem extends Vue {
    @Prop({ default: null })
    public readonly itemData: Bucket;
    @Prop({ default: () => () => {} })
    public readonly showDeleteBucketPopup: () => void;
    @Prop({ default: () => () => {} })
    public readonly openDropdown;
    @Prop({ default: false })
    public readonly isDropdownOpen: boolean;
    @Prop({ default: -1 })
    public readonly dropdownKey: number;

    public errorMessage = '';

    /**
     * Returns formatted date.
     */
    public get formattedDate(): string | undefined {
        return this.itemData.since.toLocaleString();
    }

    /**
     * Closes dropdown.
     */
    public closeDropdown(): void {
        this.openDropdown(-1);
    }

    /**
     * Holds on delete click logic.
     */
    public onDeleteClick(): void {
        this.showDeleteBucketPopup();
        this.closeDropdown();
    }

    /**
     * Redirects to bucket details page.
     */
    public onDetailsClick(): void {
        this.$router.push({
            name: RouteConfig.Buckets.with(RouteConfig.BucketsDetails).name,
            params: { bucketName: this.itemData.name },
        });
        this.closeDropdown();
    }
}
</script>

<style scoped lang="scss">
    .bucket-item {
        font-family: 'font_regular', sans-serif;
        font-style: normal;
        display: flex;
        align-items: center;
        padding: 25px 20px;
        width: calc(100% - 40px);
        font-weight: normal;
        font-size: 14px;
        line-height: 19px;
        color: #1b2533;
        cursor: pointer;

        &__name {
            display: flex;
            align-items: center;
            width: 70%;

            &__value {
                margin: 0 0 0 17px;
            }
        }

        &__date {
            width: 30%;
            margin: 0;
        }

        &__functional {
            padding: 0 10px;
            position: relative;
            cursor: pointer;

            &__dropdown {
                position: absolute;
                top: 25px;
                right: 15px;
                background: #fff;
                box-shadow: 0 20px 34px rgb(10 27 44 / 28%);
                border-radius: 6px;
                width: 255px;
                padding: 10px 0;
                z-index: 100;

                &__item {
                    display: flex;
                    align-items: center;
                    padding: 20px 25px;
                    width: calc(100% - 50px);

                    &__label {
                        margin: 0 0 0 10px;
                    }

                    &:hover {
                        background-color: #f4f5f7;
                        font-family: 'font_medium', sans-serif;

                        svg ::v-deep path {
                            fill: #0068dc;
                            stroke: #0068dc;
                        }
                    }
                }
            }
        }

        &:hover {
            background-color: #e6e9ef;
        }
    }
</style>
