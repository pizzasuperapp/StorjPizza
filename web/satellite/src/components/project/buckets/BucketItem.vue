// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="container">
        <p class="container__item name" :title="itemData.name">{{ itemData.name }}</p>
        <p class="container__item">{{ itemData.storage.toFixed(2) }}GB</p>
        <p class="container__item">{{ itemData.egress.toFixed(2) }}GB</p>
        <p class="container__item">{{ itemData.objectCount.toString() }}</p>
        <p class="container__item">{{ itemData.segmentCount.toString() }}</p>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { Bucket } from '@/types/buckets';

// @vue/component
@Component
export default class BucketItem extends Vue {
    @Prop({default: () => new Bucket('', 0, 0, 0, 0, new Date(), new Date())})
    private readonly itemData: Bucket;
}
</script>

<style scoped lang="scss">
    .container {
        padding: 20px 40px;
        outline: none;
        display: flex;
        background: #fff;
        margin-bottom: 1px;
        width: calc(100% - 80px);

        &__item {
            width: 20%;
            font-family: 'font_medium', sans-serif;
            font-size: 16px;
            margin: 0;
        }
    }

    .name {
        width: 40%;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
</style>
