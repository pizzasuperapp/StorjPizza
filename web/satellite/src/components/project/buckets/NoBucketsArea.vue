// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="no-buckets-area">
        <img class="no-buckets-area__image" src="@/../static/images/buckets/bucket.png" alt="bucket image">
        <h2 class="no-buckets-area__message">Create your first bucket to get started.</h2>
        <VButton
            label="Get Started"
            width="156px"
            height="47px"
            :on-press="navigateToWelcomeScreen"
        />
        <a
            class="no-buckets-area__second-button"
            href="https://docs.storj.io/"
            target="_blank"
            rel="noopener noreferrer"
        >
            Visit the Docs
        </a>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import { RouteConfig } from "@/router";

import VButton from "@/components/common/VButton.vue";

import { AnalyticsHttpApi } from '@/api/analytics';

// @vue/component
@Component({
    components: {
        VButton,
    }
})
export default class NoBucketArea extends Vue {

    private readonly analytics: AnalyticsHttpApi = new AnalyticsHttpApi();

    /**
     * Navigates user to welcome screen.
     */
    public navigateToWelcomeScreen(): void {
        this.analytics.pageVisit(RouteConfig.OnboardingTour.with(RouteConfig.OverviewStep).path);
        this.$router.push(RouteConfig.OnboardingTour.with(RouteConfig.OverviewStep).path);
    }
}
</script>

<style scoped lang="scss">
    h1,
    h2 {
        margin: 0;
    }

    .no-buckets-area {
        padding: 60px 0;
        font-family: 'font_regular', sans-serif;
        background-color: #fff;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        border-radius: 6px;

        &__image {
            margin-right: 10px;
        }

        &__message {
            font-family: 'font_bold', sans-serif;
            font-size: 18px;
            line-height: 26px;
            color: #354049;
            margin: 15px 0 30px;
        }

        &__second-button {
            display: flex;
            align-items: center;
            justify-content: center;
            width: 156px;
            height: 47px;
            font-size: 15px;
            line-height: 22px;
            border-radius: 6px;
            background-color: #fff;
            color: #2683ff;
            border: 1px solid #2683ff;
            margin-top: 7px;

            &:hover {
                background-color: #2683ff;
                color: #fff;
            }
        }

        &__help {
            margin-top: 20px;
            font-size: 13px;
            line-height: 18px;
            text-decoration: underline;
            color: #2683ff;
        }
    }
</style>
