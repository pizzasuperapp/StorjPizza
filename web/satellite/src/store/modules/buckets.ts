// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { Bucket, BucketCursor, BucketPage, BucketsApi } from '@/types/buckets';
import {StoreModule} from "@/types/store";

export const BUCKET_ACTIONS = {
    FETCH: 'setBuckets',
    FETCH_ALL_BUCKET_NAMES: 'getAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    CLEAR: 'clearBuckets',
};

export const BUCKET_MUTATIONS = {
    SET: 'setBuckets',
    SET_ALL_BUCKET_NAMES: 'setAllBucketNames',
    SET_SEARCH: 'setBucketSearch',
    SET_PAGE: 'setBucketPage',
    CLEAR: 'clearBuckets',
};

const {
    FETCH,
    FETCH_ALL_BUCKET_NAMES,
} = BUCKET_ACTIONS;
const {
    SET,
    SET_ALL_BUCKET_NAMES,
    SET_PAGE,
    SET_SEARCH,
    CLEAR,
} = BUCKET_MUTATIONS;
const bucketPageLimit = 7;
const firstPage = 1;

export class BucketsState {
    public allBucketNames: string[] = [];
    public cursor: BucketCursor = { limit: bucketPageLimit, search: '', page: firstPage };
    public page: BucketPage = { buckets: new Array<Bucket>(), currentPage: 1, pageCount: 1, offset: 0, limit: bucketPageLimit, search: '', totalCount: 0 };
}

interface BucketsContext {
    state: BucketsState
    commit: (string, ...unknown) => void
    rootGetters: {
        selectedProject: {
            id: string
        }
    }
}

/**
 * creates buckets module with all dependencies
 *
 * @param api - buckets api
 */
export function makeBucketsModule(api: BucketsApi): StoreModule<BucketsState, BucketsContext> {
    return {
        state: new BucketsState(),

        mutations: {
            [SET](state: BucketsState, page: BucketPage) {
                state.page = page;
            },
            [SET_ALL_BUCKET_NAMES](state: BucketsState, allBucketNames: string[]) {
                state.allBucketNames = allBucketNames;
            },
            [SET_PAGE](state: BucketsState, page: number) {
                state.cursor.page = page;
            },
            [SET_SEARCH](state: BucketsState, search: string) {
                state.cursor.search = search;
            },
            [CLEAR](state: BucketsState) {
                state.allBucketNames = [];
                state.cursor = new BucketCursor('', bucketPageLimit, firstPage);
                state.page = new BucketPage([], '', bucketPageLimit, 0, 1, 1, 0);
            },
        },
        actions: {
            [FETCH]: async function({commit, rootGetters, state}: BucketsContext, page: number): Promise<BucketPage> {
                const projectID = rootGetters.selectedProject.id;
                const before = new Date();
                state.cursor.page = page;

                commit(SET_PAGE, page);

                const result: BucketPage = await api.get(projectID, before, state.cursor);

                commit(SET, result);

                return result;
            },
            [FETCH_ALL_BUCKET_NAMES]: async function({commit, rootGetters}: BucketsContext): Promise<string[]> {
                const result: string[] = await api.getAllBucketNames(rootGetters.selectedProject.id);

                commit(SET_ALL_BUCKET_NAMES, result);

                return result;
            },
            [BUCKET_ACTIONS.SET_SEARCH]: function({commit}: BucketsContext, search: string) {
                commit(SET_SEARCH, search);
            },
            [BUCKET_ACTIONS.CLEAR]: function({commit}: BucketsContext) {
                commit(CLEAR);
            },
        },
        getters: {
            page: (state: BucketsState): BucketPage => state.page,
            cursor: (state: BucketsState): BucketCursor => state.cursor,
        },
    };
}
