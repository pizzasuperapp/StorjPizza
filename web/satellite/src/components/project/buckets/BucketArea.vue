// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="buckets-area">
        <NoBucketArea v-if="isNoBucketAreaShown" />
        <div v-else class="buckets-overflow">
            <div class="buckets-header">
                <p class="buckets-header__title">Usage per bucket</p>
                <VHeader
                    class="buckets-header-component"
                    placeholder="Buckets"
                    :search="fetch"
                />
            </div>
            <div v-if="buckets.length" class="buckets-container">
                <SortingHeader />
                <VList
                    class="buckets-list"
                    :data-set="buckets"
                    :item-component="itemComponent"
                    :on-item-click="doNothing"
                />
            </div>
            <div v-if="isEmptySearchResultShown" class="empty-search-result-area">
                <h1 class="empty-search-result-area__title">No results found</h1>
            </div>
        </div>
        <div v-if="isPaginationShown" class="buckets-area__pagination-area">
            <VPagination
                :total-page-count="totalPageCount"
                :on-page-click-callback="onPageClick"
            />
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VHeader from '@/components/common/VHeader.vue';
import VList from '@/components/common/VList.vue';
import VPagination from '@/components/common/VPagination.vue';
import BucketItem from '@/components/project/buckets/BucketItem.vue';
import NoBucketArea from '@/components/project/buckets/NoBucketsArea.vue';
import SortingHeader from '@/components/project/buckets/SortingHeader.vue';

import { BUCKET_ACTIONS } from '@/store/modules/buckets';
import { Bucket } from '@/types/buckets';

import { AnalyticsHttpApi } from '@/api/analytics';
import { AnalyticsEvent } from '@/utils/constants/analyticsEventNames';

const {FETCH, SET_SEARCH} = BUCKET_ACTIONS;

// @vue/component
@Component({
    components: {
        SortingHeader,
        NoBucketArea,
        VHeader,
        VPagination,
        VList,
    },
})
export default class BucketArea extends Vue {
    private readonly analytics: AnalyticsHttpApi = new AnalyticsHttpApi();
    /**
     * Lifecycle hook before component destruction where buckets search query is cleared.
     */
    public async beforeDestroy(): Promise<void> {
        await this.$store.dispatch(SET_SEARCH, '');
    }

    /**
     * Mock function for buckets list.
     */
    public doNothing(): void {
        // this method is used to mock prop function of common List
    }

    /**
     * Returns buckets total page count.
     */
    public get totalPageCount(): number {
        return this.$store.state.bucketUsageModule.page.pageCount;
    }

    /**
     * Returns buckets total count.
     */
    public get totalCount(): number {
        return this.$store.getters.page.totalCount;
    }

    /**
     * Returns BucketItem for common list.
     */
    public get itemComponent(): typeof BucketItem {
        return BucketItem;
    }

    /**
     * Returns buckets list of current page.
     */
    public get buckets(): Bucket[] {
        return this.$store.getters.page.buckets;
    }

    /**
     * Returns buckets search query.
     */
    public get search(): string {
        return this.$store.getters.cursor.search;
    }

    /**
     * Indicates if no bucket area is shown.
     */
    public get isNoBucketAreaShown(): boolean {
        return !this.totalCount && !this.search;
    }

    /**
     * Indicates if pagination is shown.
     */
    public get isPaginationShown(): boolean {
        return this.totalPageCount > 1;
    }

    /**
     * Indicates if empty bucket search is shown.
     */
    public get isEmptySearchResultShown(): boolean {
        return !!(!this.totalPageCount && this.search);
    }

    /**
     * Fetches buckets depending on search query.
     */
    public async fetch(searchQuery: string): Promise<void> {
        await this.$store.dispatch(SET_SEARCH, searchQuery);
        await this.analytics.eventTriggered(AnalyticsEvent.SEARCH_BUCKETS);

        try {
            await this.$store.dispatch(FETCH, 1);
        } catch (error) {
            await this.$notify.error(`Unable to fetch buckets: ${error.message}`);
        }
    }

    /**
     * Fetches buckets depends on page index.
     */
    public async onPageClick(page: number): Promise<void> {
        try {
            await this.$store.dispatch(FETCH, page);
        } catch (error) {
            await this.$notify.error(`Unable to fetch buckets: ${error.message}`);
        }
    }
}
</script>

<style scoped lang="scss">
    .buckets-area {
        margin-top: 30px;
        position: relative;

        &__pagination-area {
            width: 100%;
            display: flex;
            align-items: center;
            justify-content: flex-start;
        }
    }

    .buckets-header {
        display: flex;
        align-items: center;
        justify-content: space-between;
        margin-bottom: 10px;

        &__title {
            margin: 0;
            font-family: 'font_bold', sans-serif;
            font-size: 16px;
            line-height: 16px;
            color: #1b2533;
            white-space: nowrap;
        }
    }

    .header-container.buckets-header-component {
        height: 55px !important;
    }

    .buckets-container {
        background-color: #fff;
        border-radius: 6px;
        padding-bottom: 20px;
    }

    .empty-search-result-area {
        display: flex;
        align-items: center;
        justify-content: center;
        padding: 20px 0;
        background-color: #fff;
        border-radius: 6px;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 32px;
            line-height: 39px;
        }
    }

    .buckets-list {
        padding-top: 20px;
    }

    ::v-deep .pagination-container {
        padding-left: 0;
    }
</style>
