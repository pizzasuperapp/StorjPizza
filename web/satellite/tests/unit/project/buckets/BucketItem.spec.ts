// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import BucketItem from '@/components/project/buckets/BucketItem.vue';

import { Bucket } from '@/types/buckets';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();

const bucket = new Bucket('name', 1, 1, 1, 1, new Date(), new Date());

describe('BucketItem.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(BucketItem, {
            localVue,
            propsData: {
                itemData: bucket,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
