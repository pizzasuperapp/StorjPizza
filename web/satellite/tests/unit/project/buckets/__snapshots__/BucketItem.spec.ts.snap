// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketItem.vue renders correctly 1`] = `
<div class="container">
  <p title="name" class="container__item name">name</p>
  <p class="container__item">1.00GB</p>
  <p class="container__item">1.00GB</p>
  <p class="container__item">1</p>
  <p class="container__item">1</p>
</div>
`;
